src/coreComponents/linearAlgebra/interfaces/petsc/PetscPreconditioner.cpp

```diff
@@ -37,96 +35,108 @@ PetscPreconditioner::~PetscPreconditioner()
   clear();
 }
 
-void CreatePetscAMG( LinearSolverParameters const & params, PC precond )
+void CreatePetscAMG( LinearSolverParameters const & params, PC const precond )
 {
   // Default options only for the moment
   GEOSX_LAI_CHECK_ERROR( PCSetType( precond, PCGAMG ) );
   GEOSX_UNUSED_VAR( params )
 
   // TODO: need someone familiar with PETSc to take a look at this
 #if 0
-  GEOSX_LAI_CHECK_ERROR( PCSetType( prec, PCHMG ) );
-  GEOSX_LAI_CHECK_ERROR( PCHMGSetInnerPCType( prec, PCGAMG ) );
+  GEOSX_LAI_CHECK_ERROR( PCSetType( precond, PCHMG ) );
```

Comment:
Not related to this PR ... but are we still planning to improve this section?

Reply:
It'd be great, but I don't have a plan. @castelletto1 has spent quite some time trying to make it work and looks like it's not at all straightforward.